### PR TITLE
refactor: refactor `DemoOptions` component

### PR DESCRIPTION
**Pull request by dschlabach**
**What changed? Why?**
This PR refactors the `DemoOptions` component away from a switch/case and to a mapping pattern. I think this is appropriate for 2 reasons:
1. Mapping is easier to read and maintain without stepping through nested logic.
2. (more importantly) it will enable the next playground feature, the ability to copy "share links". This requires us to know _which options_ are associated with _which components_, and this is much easier to do with an accessible mapping available.

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: playground/nextjs-app-router/components/form/checkout-options.tsx [hunk 1]
```diff
@@ -23,78 +23,78 @@ export function CheckoutOptions() {
...
-                setProductOptions({
-                  ...productOptions,
-                  chargeId: e.target.value,
-                });
-                setCheckoutOptions?.({
-                  ...checkoutOptions,
-                  chargeId: e.target.value,
-                });
-              }}
+  return activeComponent === OnchainKitComponent.Checkout ? (
```
Comment (dschlabach) [inline playground/nextjs-app-router/components/form/checkout-options.tsx:26]: This is required so the component doesn't return `false` in any case.

File: playground/nextjs-app-router/components/form/transaction-options.tsx [hunk 2]
```diff
@@ -17,40 +17,38 @@ export function TransactionOptions() {
...
-              Calls Callback
-            </SelectItem>
-            <SelectItem value={TransactionTypes.ContractsCallback}>
-              Contracts Callback
-            </SelectItem>
-          </SelectContent>
-        </Select>
-      </div>
-    )
+  return activeComponent === OnchainKitComponent.Transaction ||
```
Comment (dschlabach) [inline playground/nextjs-app-router/components/form/transaction-options.tsx:20]: This is required so the component doesn't return `false` in any case.

**Review by brendan-defi — APPROVED** [review]

File: playground/nextjs-app-router/components/form/checkout-options.tsx (hunk 1 above)
Comment (alessey) [reply playground/nextjs-app-router/components/form/checkout-options.tsx:26]: if we have to have these checks for each component specific option is there any reason not to just include them all?

Comment (dschlabach) [reply playground/nextjs-app-router/components/form/checkout-options.tsx]: Yeah i'm actually not sure why this was included in the first place

**Review by alessey — APPROVED** [review]

**Pull request merged**